src/liboslexec/llvm_gen.cpp

```diff
@@ -1596,6 +1596,26 @@ LLVMGEN (llvm_gen_transform)
 
 
 
+// transformc (string fromspace, string tospace, color p)
+LLVMGEN (llvm_gen_transformc)
+{
+    Opcode &op (rop.inst()->ops()[opnum]);
+    ASSERT (op.nargs() == 4);
+    Symbol *Result = rop.opargsym (op, 0);
+    Symbol *From = rop.opargsym (op, 1);
+    Symbol *To = rop.opargsym (op, 2);
+    Symbol *C = rop.opargsym (op, 3);
+
+    llvm::Value *args[] = { rop.sg_void_ptr(),
+        rop.llvm_void_ptr(*C), rop.ll.constant(C->has_derivs()),
+        rop.llvm_void_ptr(*Result), rop.ll.constant(Result->has_derivs()),
```

Comment:
Maybe add a warning if `Result->has_derivs()` is true? Just so that we have breadcrumbs to follow when things go wrong many years from now and we have all forgotten about the shortcut?

Reply:
How about... not years, I'll do it this week? I just wanted to get this stuff to production for a high-priority request. But I don't know how to do it for the OCIO case except by finite differences.

Reply:
Are you thinking of handling the HSL/HSV cases with real dual2 at least? Those are the cases that are more likely to end up needing derivatives and would be harder to get right in a finite differencing setup.

I'd honestly be fine with making it a warning (error?) too. It does seem highly unlikely for derivs to be needed here (although more unlikely scenarios have come up before ...).

Reply:
Yeah, I was going to use dual2 for all the "hard coded" ones.

Reply:
I don't want to think about how to do the warning (at JIT time? or shader runtime?). I'm inclined to commit this in its current state so we can get the builds done that we need, and then immediately go fix it and make it work properly with derivs, within the next few days.

Reply:
I was thinking JIT time, but your plan sounds good.


LGTM!